http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieClusterMonitor.scala

```diff
@@ -52,86 +52,89 @@ class ZombieClusterMonitor(config: ZombieClusterConfig,
-                logger.debug(s"Project ${project.value} containing ${clusters.size} clusters is inactive in Google")
-                Future.successful(clusters.toList)
-            }
+                case false =>
+                  // If the project is inactive, all clusters in the project are zombies
+                  IO.pure(clusters.toList)
+              }
+            )
         }
-      }
-
-      // Error out each detected zombie cluster
-      zombieClusters.flatMap { cs =>
-        logger.info(s"Detected ${cs.size} zombie clusters across ${cs.map(_.googleProject).toSet.size} projects.")
-        cs.traverse { cluster =>
-          handleZombieCluster(cluster).unsafeToFuture()
-        }
-      }
-
+        // Error out each detected zombie cluster
+        _ <- zombies.parTraverse(zombie => semaphore.withPermit(handleZombieCluster(zombie)))
+        end <- IO(Instant.now)
+        duration = Duration.between(start, end)
+        _ <- logger.info(
+          s"Detected ${zombies.size} zombie clusters in ${zombies.map(_.googleProject).toSet.size} projects. Elapsed time = ${duration.getSeconds} seconds"
+        )
+      } yield ()).unsafeRunSync
   }
 
-  private def getActiveClustersFromDatabase: IO[Map[GoogleProject, Seq[Cluster]]] =
-    dbRef.inTransaction {
-      clusterQuery.listActiveWithLabels.map { clusters =>
-        clusters.groupBy(_.googleProject)
-      }
-    }
-
-  private def isProjectActiveInGoogle(googleProject: GoogleProject): Future[Boolean] =
+  private def isProjectActiveInGoogle(googleProject: GoogleProject): IO[Boolean] = {
     // Check the project and its billing info
-    (googleProjectDAO.isProjectActive(googleProject.value), googleProjectDAO.isBillingActive(googleProject.value))
-      .mapN(_ && _)
-      .recover {
-        //if we fail because of a permission error, we consider the cluster a zombie, as the permissions have been clean-up elsewhere
-        //this occurs in the case of free credits projects, which are managed elsewhere
-        case e: GoogleJsonResponseException if e.getStatusCode == 403 =>
-          logger.info(
-            s"Unable to check status of project ${googleProject.value} for zombie cluster detection due to a 403 from google. We are assuming this is a free credits project that has been cleaned up, and zombifying",
-            e
+    val res = for {
+      isProjectActive <- IO.fromFuture(IO(googleProjectDAO.isProjectActive(googleProject.value)))
+      isBillingActive <- IO.fromFuture(IO(googleProjectDAO.isBillingActive(googleProject.value)))
+    } yield isProjectActive && isBillingActive
+
+    res.recoverWith {
+      case e: GoogleJsonResponseException if e.getStatusCode == 403 =>
+        logger
+          .info(e)(
+            s"Unable to check status of project ${googleProject.value} for zombie cluster detection " +
+              s"due to a 403 from google. We are assuming this is a free credits project that has been cleaned up, and zombifying"
```

Comment:
Would the error messages above and below be misleading if it were the `isBillingActive()` that returns an exception?